[BUG] 1.6.0: build fails with latest gcc

**Describe the bug**
Looks like buiod fails with latest gcc.

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Linux/x86_64
  - Compiler and version: gcc 11.2.1

```console
[ 81%] Building CXX object test/CMakeFiles/reporter_output_test.dir/reporter_output_test.cc.o
cd /home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/x86_64-redhat-linux-gnu/test && /usr/bin/g++ -DHAVE_POSIX_REGEX -DHAVE_STD_REGEX -DHAVE_STEADY_CLOCK -DTEST_BENCHMARK_LIBRARY_HAS_NO_ASSERTIONS -I/home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/include -I/home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/src/../include -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none  -std=c++11  -Wall  -Wextra  -Wshadow  -pedantic  -pedantic-errors  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing   -Werror  -Wno-deprecated   -UNDEBUG -MD -MT test/CMakeFiles/reporter_output_test.dir/reporter_output_test.cc.o -MF CMakeFiles/reporter_output_test.dir/reporter_output_test.cc.o.d -o CMakeFiles/reporter_output_test.dir/reporter_output_test.cc.o -c /home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/test/reporter_output_test.cc
/home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/test/reporter_output_test.cc: In lambda function:
/home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/test/reporter_output_test.cc:946:66: error: unused parameter ‘v’ [-Werror=unused-parameter]
  946 | const auto UserPercentStatistics = [](const std::vector<double>& v) {
      |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
cc1plus: all warnings being treated as errors
make[2]: *** [test/CMakeFiles/reporter_output_test.dir/build.make:79: test/CMakeFiles/reporter_output_test.dir/reporter_output_test.cc.o] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/benchmark-1.6.0/x86_64-redhat-linux-gnu'
```

--- a/test/reporter_output_test.cc
+++ b/test/reporter_output_test.cc
@@ -943,7 +943,7 @@ ADD_CASES(
 // ------------- Testing relative standard deviation statistics ------------ //
 // ========================================================================= //
 
-const auto UserPercentStatistics = [](const std::vector<double>& v) {
+const auto UserPercentStatistics = [](const std::vector<double>&) {
   return 1. / 100.;
 };
 void BM_UserPercentStats(benchmark::State& state) {

